unit-tests/src/test/java/org/eclipse/collections/impl/factory/SetsTest.java

```diff
@@ -762,6 +762,21 @@ public void cartesianProduct()
     }
 
     @Test
+    public void cartesianProductWithFunction()
+    {
+        MutableSet<Integer> set1 = UnifiedSet.newSetWith(1, 2);
+        MutableSet<Integer> set2 = UnifiedSet.newSetWith(2, 3, 4);
+        MutableBag<Pair<Integer, Integer>> expectedCartesianProduct = Bags.mutable.of(
+                Tuples.pair(1, 2),
+                Tuples.pair(2, 2),
+                Tuples.pair(1, 3),
+                Tuples.pair(2, 3),
+                Tuples.pair(1, 4),
+                Tuples.pair(2, 4));
+        Assert.assertEquals(expectedCartesianProduct, Sets.cartesianProduct(set1, set2, Tuples::pair).toBag());
```

Comment:
Can you please add 3 more tests:
1) Same 2 elements in both tests
2) set2.size() > set1.size()
3) Something other than Tuples::pair, it can even be first + second